### PR TITLE
add warning when user misconfigures configurator networks

### LINKED ISSUE #844: Warn the user he is configuring multiple interfaces in the same network
As this seems a recurring question on the mailing list and here.

Better display a little warning to the user so he knows

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Warn the user when he is configuring multiple interfaces in the same network in the configurator as it seems to be a recurring issue
# Impacts

Less 'chair/keyboard' github issues and mailing list posts
# Issue

fixes #844 
# Delete branch after merge

YES
# NEWS file entries
## Enhancements
- Added a warning in the configurator when the user is configuring multiple interfaces in the same network

### CODE REVIEW

File: html/pfappserver/lib/pfappserver/Model/Interface.pm [hunk 1]
```diff
@@ -746,6 +746,31 @@ sub getEnforcement {
     return $enforcement;
 }
 
+=head2 sort_interfaces_by_network
+
+Will create a hash that maps which interfaces are tied to which network
+
+=cut
+
+sub sort_interfaces_by_network {
```
Comment (louismunro) [inline html/pfappserver/lib/pfappserver/Model/Interface.pm]: Wrong name? 
map_interface_to_networks?

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Configurator.pm [hunk 2]
```diff
@@ -242,7 +242,8 @@ sub networks :Chained('object') :PathPart('networks') :Args(0) {
         $c->session->{gateway} = $c->model('Config::System')->getDefaultGateway if (!defined($c->session->{gateway}));
 
         my $interfaces_ref = $c->model('Interface')->get('all');
-        $c->stash(interfaces => $interfaces_ref);
+        $c->stash->{interfaces} = $interfaces_ref;
+        $c->stash->{seen_networks} = $c->model('Interface')->sort_interfaces_by_network($c->stash->{interfaces});
```
Comment (louismunro) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Configurator.pm]: Change name here too.

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Interface.pm [hunk 3]
```diff
@@ -77,6 +77,9 @@ sub list :Local :Args(0) :AdminRole('INTERFACES_READ') {
     my ( $self, $c ) = @_;
 
     $c->stash->{interfaces} = $c->model('Interface')->get('all');
+
+    $c->stash->{seen_networks} = $c->model('Interface')->sort_interfaces_by_network($c->stash->{interfaces});
+
```
Comment (louismunro) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Interface.pm:82]: and here.

**Comment by julsemaan** [comment]
Re-review

**Pull request merged**